test/e2e/utils_test.go

```diff
@@ -108,32 +108,19 @@ func exposeAdminAPI(ctx context.Context, t *testing.T, env environments.Environm
 // returns the modified manifest path. If there is any issue patching the manifest, it will log the issue and return
 // the original provided path.
 func getTestManifest(t *testing.T, baseManifestPath string) (io.Reader, error) {
-	var manifestsReader io.Reader
-	manifestsReader, err := os.Open(baseManifestPath)
+	var (
+		manifestsReader io.Reader
+		err             error
+	)
+	manifestsReader, err = os.Open(baseManifestPath)
 	if err != nil {
 		return nil, err
 	}
 
-	var imageFullname string
-	if imageLoad != "" {
-		imageFullname = imageLoad
-	} else {
-		imageFullname = imageOverride
-	}
-
-	if imageFullname != "" {
-		split := strings.Split(imageFullname, ":")
-		if len(split) < 2 {
-			t.Logf("could not parse override image '%v', using default manifest %v", imageFullname, baseManifestPath)
-			return manifestsReader, nil
-		}
-		repo := strings.Join(split[0:len(split)-1], ":")
-		tag := split[len(split)-1]
-		manifestsReader, err = patchControllerImage(manifestsReader, repo, tag)
-		if err != nil {
-			t.Logf("failed patching override image '%v' (%v), using default manifest %v", imageFullname, err, baseManifestPath)
-			return manifestsReader, nil
-		}
+	manifestsReader, err = patchControllerImageHelper(manifestsReader, baseManifestPath)
+	if err != nil {
+		t.Logf("failed patching controller image (%v), using default manifest %v", err, baseManifestPath)
```

Comment:
Shouldn't this be a test failure? I get it just repeats the logic that was here before, but IMO we should be explicit in such cases as we may end up with tests that do not do what we expect them to do (I guess it's easy to not notice this log). WDYT?

Reply:
Let's leave it as it is. We can create an issue to track the effort to improve this. There might have been a reason (🤷) to make it this way.